core/src/main/java/org/mule/internal/connection/PoolingConnectionHandlingStrategy.java

```diff
@@ -70,7 +72,30 @@
     {
         try
         {
-            return new PooledConnectionHandler<>(config, borrowConnection(), pool, poolingListener);
+            boolean isValid = false;
+            Connection connection = null;
+
+            while (!isValid)
```

Comment:
Not sure a while is a good idea... If the endpoint is down this can turn into an infinite loop. I'd rather just throw a ConnctionException and let the reconnection thingy take over